pkg/openstack/placement.go

```diff
@@ -32,9 +38,63 @@ func ReconcilePlacementAPI(ctx context.Context, instance *corev1beta1.OpenStackC
 		return ctrl.Result{}, nil
 	}
 
+	// add selector to service overrides
+	for _, endpointType := range []service.Endpoint{service.EndpointPublic, service.EndpointInternal} {
+		if instance.Spec.Placement.Template.Override.Service == nil {
+			instance.Spec.Placement.Template.Override.Service = map[string]service.OverrideSpec{}
+		}
+		instance.Spec.Placement.Template.Override.Service[string(endpointType)] = AddServiceComponentLabel(
+			ptr.To(instance.Spec.Placement.Template.Override.Service[string(endpointType)]),
+			placementAPI.Name)
+	}
+
+	// When component services got created check if there is the need to create a route
+	if err := helper.GetClient().Get(ctx, types.NamespacedName{Name: "placement", Namespace: instance.Namespace}, placementAPI); err != nil {
+		if !k8s_errors.IsNotFound(err) {
+			return ctrl.Result{}, err
+		}
+	}
+
+	// When component services got created check if there is the need to create a route
+	if err := helper.GetClient().Get(ctx, types.NamespacedName{Name: "placement", Namespace: instance.Namespace}, placementAPI); err != nil {
+		if !k8s_errors.IsNotFound(err) {
+			return ctrl.Result{}, err
+		}
+	}
+
+	if placementAPI.Status.Conditions.IsTrue(condition.ExposeServiceReadyCondition) {
+		svcs, err := service.GetServicesListWithLabel(
+			ctx,
+			helper,
+			instance.Namespace,
+			map[string]string{common.AppSelector: placementAPI.Name},
+		)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		var ctrlResult reconcile.Result
+		instance.Spec.Placement.Template.Override.Service, ctrlResult, err = EnsureRoute(
+			ctx,
+			instance,
+			helper,
+			placementAPI,
```

Comment:
I'm wondering if we need that the service CR owns the Route. I don't really see a technical issue with this now just feels like we are reaching over the head of the service operator and forcing it to "adopt" a route. What are the pros of having the owner reference pointing to the service CR instead of simply letting the OpenstackControlPlane CR to own the Route it creates.

Reply:
the ctlplane CR still owns it as the controller, as mentioned above we just add the service CR to the ownerref list so that we block early deletion and make sure it stays until the service cr is gone

Reply:
I see. Is there a specific need for the route to outlive the OpenStackControlPlane CR? I guess the PlacementAPI only owner is also the  OpenStackControlPlane CR? So during OpenStackControlPlane deletion both will be deleted anyhow.

Reply:
the case where this is required is for keystone. when the osctlplane CR gets deleted, it might immediately start cleaning the resources owned by the osctlplane. when the route for keystone is gone, the cleanup of  services/endpoints might fail. Therefore the routes should be cleaned after the service is gone.

Reply:
Ohh. Where do we rely on the public keystone endpoint to clean up resources? That seems like a bug. I would assume that the operators use the internal keystone endpoint to do the cleanup.

Reply:
the admin service client from [1] is using the public endpoint. its probably ok to switch that to internal

[1] https://github.com/openstack-k8s-operators/keystone-operator/blob/main/api/v1beta1/keystoneapi.go#L78

Reply:
I would switch [1] to internal just to be on the safe side. But obviously that is a separate change. 
I have no hard opinion about whether to keep or remove the owner ref.